src/main/java/com/networknt/schema/RefValidator.java

```diff
@@ -66,16 +63,47 @@ public RefValidator(String schemaPath, JsonNode schemaNode, JsonSchema parentSch
                 schema = parentSchema.findAncestor();
             } else {
                 refValue = refValue.substring(index);
-                if (refValue.equals("#")) {
-                    schema = parentSchema.findAncestor();
-                } else {
-                    JsonNode node = parentSchema.getRefSchemaNode(refValue);
-                    if (node != null) {
-                        schema = new JsonSchema(mapper, refValue, node, parentSchema);
-                    }
-                }
             }
         }
+        if (refValue.equals(REF_CURRENT)) {
+            schema = parentSchema.findAncestor();
+        } else {
+            JsonNode node = parentSchema.getRefSchemaNode(refValue);
+            if (node != null) {
+                schema = new JsonSchema(mapper, refValue, node, parentSchema);
+            }
+        }
+    }
+    
+    private boolean isRelativePath(String schemaUrl) {
+    	return !schemaUrl.startsWith("http");
+    }
+    
+    private String obtainAbsolutePath(JsonSchema parentSchema, String schemaUrl) {
+    	String baseSchemaUrl = parentSchema.findAncestor().getSchemaNode().get("id").textValue();

```

Comment:
Where does this `"id"` property come from and why is it treated as a URL?

Reply:
I found https://spacetelescope.github.io/understanding-json-schema/structuring.html?highlight=dependencies#the-id-property. But it should be optional, I'd say. I still want to use $ref on local relative files.